lib/gridfs-stream/download.js

```diff
@@ -278,52 +278,56 @@ function init(self) {
     if (error) {
       return __handleError(self, error);
     }
-    if (!doc) {
-      var identifier = self.s.filter._id ? self.s.filter._id.toString() : self.s.filter.filename;
-      var errmsg = 'FileNotFound: file ' + identifier + ' was not found';
-      var err = new Error(errmsg);
-      err.code = 'ENOENT';
-      return __handleError(self, err);
-    }
+    try {
```

Comment:
this feels like a bit of a sledgehammer approach, is there a way to satisfy the issues in the ticket without wrapping the world in a try/catch?

Reply:
Well, I could specifically wrap the calls to `handleStartOptions` and `handleEndOptions` in try/catch, but this approach seemed more robust. What makes it a "sledgehammer approach?" I'm open to other ideas.